Assets/PatchKit Patcher/Scripts/AppData/Local/Pack1Unarchiver.cs

```diff
@@ -68,35 +82,68 @@ public void Unarchive(CancellationToken cancellationToken)
             DebugLogger.Log("Unpacking " + _metaData.Files.Length + " files...");
             foreach (var file in _metaData.Files)
             {
-                OnUnarchiveProgressChanged(file.Name, file.Type == "regular", entry, _metaData.Files.Length, 0.0);
+                OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, entry, _metaData.Files.Length, 0.0);
 
                 var currentFile = file;
                 var currentEntry = entry;
-                Unpack(file, progress =>
+
+                if (CanUnpack(file))
                 {
-                    OnUnarchiveProgressChanged(currentFile.Name, currentFile.Type == "regular", currentEntry, _metaData.Files.Length, progress);
-                });
+                    Unpack(file, progress =>
+                    {
+                        OnUnarchiveProgressChanged(currentFile.Name, currentFile.Type == Pack1Meta.regularFileType, currentEntry, _metaData.Files.Length, progress);
+                    });
+                }
 
-                OnUnarchiveProgressChanged(file.Name, file.Type == "regular", entry, _metaData.Files.Length, 1.0);
+                OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, entry, _metaData.Files.Length, 1.0);
 
                 entry++;
             }
             DebugLogger.Log("Unpacking finished succesfully!");
         }
 
+        public void UnarchiveSingleFile(Pack1Meta.FileEntry file, CancellationToken cancellationToken)
+        {
+            OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, 0, 1, 0.0);
+
+            if (!CanUnpack(file))
+            {
+                throw new ArgumentOutOfRangeException("file", file, null);
+            }
+
+            Unpack(file, progress => OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, 1, 1, progress));
+
+            OnUnarchiveProgressChanged(file.Name, file.Type == Pack1Meta.regularFileType, 0, 1, 1.0);
+        }
+
+        private bool CanUnpack(Pack1Meta.FileEntry file)
+        {
+            if (file.Type != Pack1Meta.regularFileType)
+            {
+                return true;
```

Comment:
Hold on, shouldn't it be `false` instead?

Reply:
Is there any situation in which the "symlink" or "directory" file type couldn't be unpacked?

Reply:
More concerning is that I don't know any situation when regular file shouldn't be unpacked.

Also 'unpacking' a directory and a symlink is practically meaningless, because those file types does not have any 'content'. Symlink just points to another file, directory is just a path.

Reply:
It's less about shouldn't and more about can't.

This method is a safeguard for a situation in which the Patcher would try to unpack a file that's not present in the partial pack1 file.

Reply:
I think that in that case it should raise an exception.